dipy/reconst/dti.py

```diff
@@ -1514,111 +1580,129 @@ def _ols_fit_matrix(design_matrix):
     return np.dot(U, U.T)
 
 
-def _nlls_err_func(tensor, design_matrix, data, weighting=None,
-                   sigma=None):
-    r"""
-    Error function for the non-linear least-squares fit of the tensor.
+class _nlls_class():
```

Comment:
Not sure to see the benefit of a class here.  is there a specific state that you want to keep? maybe a closure is enough. Can you explain ? thank you

Reply:
I'd name it so that it's clear that it's a class in subsequent usage (when it's initialized further down). How about: `_NllsHelper`?